src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -253,7 +253,25 @@ public sealed class ActivityListener : IDisposable
         public System.Diagnostics.SampleActivity<string>? SampleUsingParentId { get { throw null; } set { throw null; } }
         public System.Diagnostics.SampleActivity<ActivityContext>? Sample { get { throw null; } set { throw null; } }
         public void Dispose() { throw null; }
-   }
+    }
+
+    public delegate bool PropagatorGetterCallback(object carrier, string fieldName, out string? value);
+
+    public abstract class TextMapPropagator
+    {
+        public abstract System.Collections.Generic.IEnumerable<string> Fields { get; }
```

Comment:
I assume this isn't `IReadOnlySet` due to this library being built for older targets?
Should it at least be `IReadOnlyList` / `IReadOnlyCollection` instead?

Reply:
I have reviewed the [design guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections?ranMID=46131&ranEAID=a1LgFw09t88&ranSiteID=a1LgFw09t88-9V3RB6wfgJFj9pKA4dp3cg&epi=a1LgFw09t88-9V3RB6wfgJFj9pKA4dp3cg&irgwc=1&OCID=AID2000142_aff_7806_1243925&tduid=%28ir__mrdukhopi0kfqxeykk0sohzifv2xutv6bnxvqk3y00%29%287806%29%281243925%29%28a1LgFw09t88-9V3RB6wfgJFj9pKA4dp3cg%29%28%29&irclickid=_mrdukhopi0kfqxeykk0sohzifv2xutv6bnxvqk3y00#collection-properties-and-return-values) and it looks `IReadOnlyCollection` would be more reasonable here. I'll change that. Thanks!